src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -441,29 +473,16 @@ private unsafe void CreateRequestQueueHandle()
                 throw new HttpListenerException(Marshal.GetLastWin32Error());
             }
 
-            _requestQueueHandle = requestQueueHandle;
+            Debug.Assert(_currentSession is null);
+            _currentSession = new HttpListenerSession(this, requestQueueHandle);
```

Comment:
Why is CloseRequestQueueHandle delegated entirely to HttpListenerSession, but CreateRequestQueueHandle isn't? That seems asymmetric.

Reply:
No real reason, I can the move Create to Session as well.

Reply:
Moved